File: foodDeliveryFrontend/client/src/service/menuItemService.js
const API_BASE_URL = "http://localhost:8089/api";

export const menuItemService = {
  createMenuItem: async (menuItemData, token) => {
    return fetch(`${API_BASE_URL}/restaurants/menu-items`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(menuItemData),
    });
  },

  getMenuItemById: async (id) => {
    return fetch(`${API_BASE_URL}/restaurants/menu-items/${id}`);
  },

  getMenuItemsByRestaurantId: async (restaurantId, token) => {
    return fetch(
      `${API_BASE_URL}/restaurants/menu-items/by-restaurant/${restaurantId}`,
      {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${token}`,
        },
      }
    );
  },

  getMenuItemsByCategoryId: async (categoryId, token) => {
    return fetch(
      `${API_BASE_URL}/restaurants/menu-items/by-category/${categoryId}`,
      {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${token}`,
        },
      }
    );
  },

  updateMenuItem: async (id, menuItemData, token) => {
    return fetch(`${API_BASE_URL}/restaurants/menu-items/${id}`, {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(menuItemData),
    });
  },

  deleteMenuItem: async (id, token) => {
    return fetch(`${API_BASE_URL}/restaurants/menu-items/${id}`, {
      method: "DELETE",
      headers: {
        Authorization: `Bearer ${token}`,
      },
    });
  },
};

File: restaurantService/src/main/java/com/foodDelivery/restaurantService/serviceInterfaces/MenuItemService.java
package com.foodDelivery.restaurantService.serviceInterfaces;

import com.foodDelivery.restaurantService.dto.MenuItemCreateRequest;
import com.foodDelivery.restaurantService.dto.MenuItemResponse;

import java.util.List;

public interface MenuItemService {
    MenuItemResponse createMenuItem(MenuItemCreateRequest request, String token);
    MenuItemResponse getMenuItemById(String id);
    List<MenuItemResponse> getMenuItemsByRestaurantId(String restaurantId);
    List<MenuItemResponse> getMenuItemsByCategoryId(String categoryId);
    MenuItemResponse updateMenuItem(String id, MenuItemCreateRequest request, String token);
    void deleteMenuItem(String id);
}